x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -190,11 +190,41 @@ export function trainedModelsRoutes(
             // we don't need to fill kibana's log with these messages.
             mlLog.debug(e);
           }
+          const enabledFeatures = getEnabledFeatures();
+          try {
+            if (enabledFeatures.dfa) {
+              const jobIds = result.map((model) => {
+                let id = model.metadata?.analytics_config?.id;
+                if (id) {
+                  id = `${id}*`;
+                }
+                return id;
+              });
+              const filteredJobIds = jobIds.filter((id) => id !== undefined);
 
-          const body = filterForEnabledFeatureModels(result, getEnabledFeatures());
+              const { data_frame_analytics: jobs } = await mlClient.getDataFrameAnalytics({
+                id: filteredJobIds.join(','),
+                allow_no_match: true,
+              });
+
+              jobs.forEach(({ id }) => {
```

Comment:
We should be looping through models, not jobs. We need to set `false` for all job ids that aren't found, and by looping through jobs only we won't know if the jobs hasn't been foun

Reply:
If the job hasn't been returned then the `origin_job_exists` property wouldn't be added to the model - though if we want to explicitly set it to 'false' when it's not found then I agree that looping through the models makes more sense. Will update it shortly.

Reply:
Updated to loop through the filtered models to ensure all dfa models get the `origin_job_exists` property set explicitly in 50ffcfdd594f6cedb13acedc195b6cd00bac950d

Reply:
>I was originally thinking that it would be more efficient to just loop through the returned jobs

I was actually thinking that to make this more efficient we could keep a list of the dfa models when identifying them in the reduce and only loop through those later on. e.g.
```ts
const dfaModels = [];
const jobIdsString = filteredModels.reduce(
  (jobIdsStr: string, currentModel: TrainedModelConfigResponse, idx: number) => {
    if (isTrainedModelConfigResponse(currentModel)) {
      dfaModels.push(currentModel);
      ....
```
But it's not needed, the time difference will be milliseconds